src/routers/router.ts

```diff
@@ -0,0 +1,202 @@
+  let networkId = req.params.networkId
+
+  database.accountCollection.findOne({ uniqueId: accountId }, (err: Error, account: any) => {
+    if (err) {
+      console.error("Error fetching account for public address subscription. " + err)
+      res.status(500).send()
+    } else {
+      if (account) {
+        let publicAddress = req.body.publicAddress
+        if (publicAddress) {
+          let publication = {
+            publicAddress: publicAddress,
+            network: networkId
+          }
+          database.publicAddressSubscriptionCollection.insert(publication, (errPUB: Error, pub: any) => {
+            if (errPUB) {
+              console.error("Error creating public address subscription. " + errPUB.message)
+            } else {
+              var updateOrSet = {}
+              if (account.publicAddressSubscriptions) {
+                updateOrSet = { $push: { publicAddressSubscriptions: pub[0]._id } }
+              } else {
+                updateOrSet = { $set: { publicAddressSubscriptions: [pub[0]._id] } }
+              }
+              database.accountCollection.update({ _id: account._id }, updateOrSet, (errUPT: Error, response: any) => {
+                if (errUPT) {
+                  console.error("Error adding public address subscription on account. " + errUPT.message)
+                  res.status(500).send()
+                } else {
+                  if (account.webPushNotification) {
+                    database.webPushNotificationCollection.findOne({ _id: account.webPushNotification }, (err: Error, wpn: any) => {
+                      if (err) {
+                        console.error("Error fetching webpush notification parameters for account " + account.uniqueId + ' ' + err.message)
+                        res.status(500).send()
+                      } else {
+                        TrackerUtils.watchWebPush(networkId, publicAddress, getWebPushSubstriptionObject(wpn))
+                        res.status(201).json(pub[0]).send()
+                      }
+                    })
+                  } else {
+                    res.status(500).send({ message: 'No web push parameters found for this account' })
+                  }
+                }
+              })
+            }
+          })
+        } else {
+          console.error('Error parsing public address subscription parameters.')
+          res.status(400).send()
+        }
+      } else {
+        res.status(404).send()
+      }
+    }
+  })
+})
+
+// Create WebPushNotification
+// /accounts/:accountId/webpush-notifications
+webpushNotificationRouter.post('/', (req: any, res: any) => {
+  let accountId = req.params.accountId
+  database.accountCollection.findOne({ uniqueId: accountId }, (err: Error, account: any) => {
+    if (err) {
+      console.error("Error occured fetching account " + accountId + ' ' + err.message)
+      res.status(500).send()
+    }
+    if (account) {
+      let webPushNotif
+      try {
+        webPushNotif = {
```

Comment:
Why not just `webPushNotif = req.body.subscription`?

Reply:
Key level is not needed. I will change the body client-side to remove key level and directly get subscription server-side